libsledge/src/memory_instructions.c

```diff
@@ -18,47 +18,89 @@ INLINE float
 get_f32(uint32_t offset)
 {
 	assert(sledge_abi__current_wasm_module_instance.memory.buffer != NULL);
-	assert((uint64_t)offset + sizeof(float) <= sledge_abi__current_wasm_module_instance.memory.size);
+
+	if (unlikely((uint64_t)offset + (uint64_t)sizeof(float)
+	             > sledge_abi__current_wasm_module_instance.memory.size)) {
+		fprintf(stderr, "get_f32 OOB: offset %u + length %u > size %lu\n", offset, (uint32_t)sizeof(float),
+		        sledge_abi__current_wasm_module_instance.memory.size);
+		sledge_abi__wasm_trap_raise(WASM_TRAP_OUT_OF_BOUNDS_LINEAR_MEMORY);
+	}
+
 	return *(float *)&sledge_abi__current_wasm_module_instance.memory.buffer[offset];
 }
 
 INLINE double
 get_f64(uint32_t offset)
 {
 	assert(sledge_abi__current_wasm_module_instance.memory.buffer != NULL);
-	assert((uint64_t)offset + sizeof(double) <= sledge_abi__current_wasm_module_instance.memory.size);
+
+	if (unlikely((uint64_t)offset + (uint64_t)sizeof(double)
+	             > sledge_abi__current_wasm_module_instance.memory.size)) {
+		fprintf(stderr, "get_f64 OOB: offset %u + length %u > size %lu\n", offset, (uint32_t)sizeof(double),
```

Comment:
Did you performance regress this? I'd wonder if the instruction locality to suffer significantly due to this exception code blowing up the binary. Any ideas on how to minimize it?